### PR TITLE
JBPM-7317 Perfomance Improvements (getShape, applyState, remove CDI Event on draw)

### PR DESCRIPTION
**Pull request by tiagodolphine**
- Get shape using a Map instead of a List on AbstractCanvas
- Remove CDI event after canvas draw
- Selection only call applyState selected/unselected elements

Related to JBPM-7267: Improve computation performance 

@romartin

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LienzoMultipleSelectionControlTest.java [hunk 1]
```diff
@@ -346,7 +345,7 @@ public void testMoveSelectionShapeToTop() {
                                                       clearSelectionEvent,
                                                       ssp);
         tested.init(canvasHandler);
-        tested.onCanvasDrawn(new CanvasDrawnEvent(canvas));
+        tested.onCanvasSelection(new CanvasSelectionEvent(canvasHandler, Collections.emptyList()));
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LienzoMultipleSelectionControlTest.java]: mmm good point, it's great removing this, it was tricky.... but not sure if only observing now the `CanvasSelectionEvent` will work, give it a good test from the UI... I mean do things like select multiple elements (via area or via shift+click) and then remove/add the selection into/from some parent,then drag the parent too... as the selection keeps there, the rectangle are it should be always on top, just double check this plz

**Comment by jomarko** [comment]
@tiagodolphine just FYI, you have one checkstyle violation.
```
[ERROR] src/main/java/org/kie/workbench/common/stunner/core/client/canvas/Canvas.java:[20,8] (imports) UnusedImports: Unused import - java.util.List.
```

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Strange, getting error `11:10:03 ERROR: ci-rhos-rhel7-xlarge-xmem-382 is offline; cannot locate jdk1.8` on jenkins.

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by LuboTerifaj** [comment]
Yeah it is strange. I will try it run again.

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest this

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Pull request merged**